### DISCUSSION TITLE
Trying to get a view collection to create a relation on json_group_array

### DISCUSSION THREAD

**Discussion by jonshipman** [pocketbase/pocketbase#4085] [Q&A] [answered]
```
SELECT
  (ROW_NUMBER() OVER()) AS id,
  p.key AS plugin,
  json_extract(p.value,'$.Name') AS name,
  json_extract(p.value,'$.Version') AS version,
  json_group_array(site.id) AS site
FROM site
LEFT JOIN json_each(site.plugins) p
WHERE site.active = TRUE
GROUP BY plugin, version
```

`site` is the collection/table with a json column named `plugins`. 

<img width="1661" alt="image" src="https://github.com/pocketbase/pocketbase/assets/12699954/edcccf87-89e9-4999-8dcd-ebe23eb88346">

The query is working fine, but `expand` does not work on the `site` column. Was wondering if this "should" work.

**Comment by jonshipman** [comment]
[collections.json](https://github.com/pocketbase/pocketbase/files/13836205/collections.json)
[plugins.json](https://github.com/pocketbase/pocketbase/files/13836220/plugins.json)

Collections and a sample of the plugins JSON field for a site.

**Comment by ganigeorgiev** [comment, marked as answer]
What you want to do is not supported at the moment because SQL expressions are treated as `json` field type and `expand` works only with `relation` fields.

There is a [task in the roadmap](https://github.com/orgs/pocketbase/projects/2/views/1?pane=issue&itemId=32302192) to reconsider allowing explicit view schema fields mapping so that users can create for example manually the `relation` field definition for the column, but I have to think a little more on it and remains a low priority for now.
